lib/validator.js

```diff
@@ -218,22 +242,30 @@ Validator.prototype.validateSchema = function validateSchema (instance, schema,
 
   if (checkProperties) {
     var skipAttributes = options && options.skipAttributes || [];
-    // Validate each schema attribute against the instance
-    for (var key in schema) {
-      if (!attribute.ignoreProperties[key] && skipAttributes.indexOf(key) < 0) {
-        var validatorErr = null;
-        var validator = self.attributes[key];
-        if (validator) {
-          validatorErr = validator.call(self, instance, schema, options, ctx);
-        } else if (options.allowUnknownAttributes === false) {
-          // This represents an error with the schema itself, not an invalid instance
-          throw new SchemaError("Unsupported attribute: " + key, schema);
-        }
-        if (validatorErr) {
-          result.importErrors(validatorErr);
-        }
+
+    // Give required:false special treatment to override other attributes when
+    // instance is an empty string
+    var requiredOverrideOtherAttributes = false;
```

Comment:
Shouldn't this be applied in `attribute.js` for `enum`, `format`, and `pattern` instead checking all and skipping all?